chalicelib/utils.py

```
@@ -183,3 +187,51 @@ def get_content_type(response):
         return response.text
 
 
+def get_data(dirname):
+    """Return dictionary of participant and sample TSV filenames. The logic
+    identifies the path to the file containing the participant information
+    and another containing the sample and other information, based on
+    their column headers.
+    :param dirname: (str) current working directory
+    :return data: (dict)"""
+
+    data = {'participant': '', 'sample': ''}
+    arr = []
+    eventflag = None  # make loop execution dependent on event flag values
+    for path, subdirs, files in os.walk(dirname):
+        for name in files:
+            if 'data' and '.tsv' in name:
+                arr.append(os.path.join(path, name))
+    if len(arr) != 2:
+        msg = 'number of TSV files is {}, must be exactly 2'.format(len(arr))
+        return Response(body={'message': msg},
```

You need to return an error status code. Since this case means that the data submitted was incorrect, meaning it is a user correctable error, you should set status code to 400.